tools/userland-fetch

```diff
@@ -383,11 +383,22 @@ def main():
 
                 # An invalid hash shouldn't cause us to remove
                 # the target file if the signature was valid.
+                # Also, if the developer is in progress of upgrading
+                # some package version or introduces a new one, and
+                # explicitly ran "gmake fetch", keep the downloaded
+                # file (Makefile is not in position to have a valid
+                # checksum entry just yet) so it does not have to be
+                # downloaded twice.
                 if not sig_valid:
-                    try:
-                        os.remove(name)
-                    except OSError:
-                        pass
+
+                    if keep_arg == False:
+                        try:
+                            print "\nWARN: Removing the corrupt downloaded file"
```

Comment:
It already does a lot of diagnostics prints above - like what it is pulling, what checksums were expected and actual etc. And it did print this line until the block was lost to refactoring a couple of commits back. And in fact knowing the verdict and action taken in the end may be as (or more) useful than the 10 or so lines printed anyway before this.

I agree that the tool could use a verbosity setting, but that is out of scope for this change and would impact many other lines of code here.

Also note that the usual use-case is of downloading or using already-cached file which has its expected checksum correctly, so this block does not fire at all.